### DISCUSSION TITLE
[Help]: Slide dynamic sizing

### DISCUSSION THREAD

**Discussion by joacoespinosa** [davidjerleke/embla-carousel#924] [Help] [answered]
My app currently displays a set of slides with different widths, and I would like to offer the user the possibility of scaling (either up or down) these widths.

I'm attempting to achieve this by dynamically modifying the flex basis property of the slides, but I unfortunately trigger some sort of flickering while doing this.

Here's a video that captures the essence of my objective and the flickering as well:

[Screencast from 2024-06-28 18-40-03.webm](https://github.com/davidjerleke/embla-carousel/assets/23172865/14ef3663-b78e-4c7d-9074-b892c9be81fe)

Questions:
1. Could this be related to the number of reInits triggered?
2. Is there a way to smoothly achieve these transformations?

Please note that I'm using `containScroll: false`

```typescript
const OPTIONS: EmblaOptionsType = { containScroll: false }
```

### CodeSandbox example

https://codesandbox.io/p/sandbox/embla-carousel-variable-widths-react-forked-6ml2c7

**Comment by joacoespinosa** [comment]
@davidjerleke Is there some documentation relating to dynamic sizing of slides? I could only find one about variable (but static) widths. Thank you for your help

**Reply by davidjerleke** [reply, marked as answer]
@joacoespinosa thanks for your question. This seems worth investigating and it might be a bug. It could be related to this:

- #905

**Comment by KikeVida** [comment]
Hi @davidjerleke, was this bug fixed? I'm having the same issue

**Reply by davidjerleke** [reply]
@KikeVida multiple devs say that the problem was solved here:

- #905

...however, if you're experiencing problems, please share a CodeSandbox with your setup and don't forget to provide steps to reproduce the problem too. Because I can't blindly guess why you're experiencing problems. Thanks for understanding.

**Reply by KikeVida** [reply]
My bad David, I realized that the project I am working on has a version of embla-carousel before you released version v8.1.8. I just tried this new version, and it worked correctly. Thank you so much for your quick response:)
